src/modules/roc_sndio/target_sox/roc_sndio/sox_sink.cpp

```diff
@@ -134,6 +139,11 @@ void SoxSink::write(audio::Frame& frame) {
 }
 
 bool SoxSink::prepare_() {
+    if (buffer_size_ == 0) {
+        out_signal_.rate = sample_rate();
```

Comment:
I guess this line is not needed:

```
out_signal_.rate = sample_rate();
```

We should only update buffer size. `out_signal_.rate` should be already correct here.

Reply:
done.